core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -348,8 +533,6 @@ public RetryOnPrimaryException(StreamInput in) throws IOException{
         private final InternalRequest internalRequest;
```

Comment:
This the first line I can comment on for this class- can we update the java doc (it doesn't do everything up to sending requests to replicas any more). 

Also in terms of naming: we used to have PrimaryPhase and ReplicaPhase  , now some of the logic is split between PrimaryPhase , AsyncPrimaryAction and ReplicaPhase.   I think it would be clearer if we have ResolvePrimaryPhase, PrimaryPhase (which is the logic in AsyncPrimaryAction) and ReplicaPhase.